src/rgw/rgw_data_sync.cc

```diff
@@ -1239,17 +1242,16 @@ class RGWDataSyncCR : public RGWCoroutine {
 
       yield {
         if  ((rgw_data_sync_info::SyncState)sync_status.sync_info.state == rgw_data_sync_info::StateSync) {
-          case rgw_data_sync_info::StateSync:
```

Comment:
@oritwas this line shouldn't have been there.. there's no switch statement, it only compiles because the reenter/yield expand into a switch statement. The `if` there replaced the original switch.